src/main/java/taxi/dao/jdbcimpl/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,210 @@
+package taxi.dao.jdbcimpl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import taxi.dao.CarDao;
+import taxi.exception.DataProcessingException;
+import taxi.lib.Dao;
+import taxi.model.Car;
+import taxi.model.Driver;
+import taxi.model.Manufacturer;
+import taxi.util.ConnectionUtil;
+
+@Dao
+public class CarDaoJdbcImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String insertQuery = "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?);";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection
+                         .prepareStatement(insertQuery, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject("GENERATED_KEY", Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not add new manufacturer: "
+                    + car.toString(), e);
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String selectQuery = "SELECT c.id AS car_id, model, "
+                + "m.manufacturer_name AS manufacturer_name,"
+                + "m.manufacturer_country AS manufacturer_country, "
+                + "m.id AS manufacturer_id "
+                + "FROM cars c "
+                + "INNER JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.id = ? AND c.deleted = FALSE;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(selectQuery)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(createCar(resultSet, connection));
```

Comment:
let's not pass the connection to the private method

let's better call `getAllDriversByCarId` after the `catch` block when our connection is already closed

Reply:
Refactored as required.